task/sensor/src/main.rs

```diff
@@ -213,32 +187,33 @@ impl idl::InOrderSensorImpl for ServerImpl {
         id: SensorId,
         nodata: NoData,
         timestamp: u64,
-    ) -> Result<(), RequestError<SensorApiError>> {
-        let r = match self.last_reading(id)? {
+    ) -> Result<(), RequestError<Infallible>> {
+        let r = match self.last_reading.get(id) {
             None | Some(LastReading::ErrorOnly) => LastReading::ErrorOnly,
             Some(
                 LastReading::Data | LastReading::DataOnly | LastReading::Error,
             ) => LastReading::Error,
         };
 
-        self.last_reading[id] = Some(r);
-        self.err_value[id] = nodata;
-        self.err_time[id] = timestamp;
+        *self.last_reading.get_mut(id) = Some(r);
+        *self.err_value.get_mut(id) = nodata;
+        *self.err_time.get_mut(id) = timestamp;
 
         //
         // We pack per-`NoData` counters into a u32.
         //
         let (nbits, shift) = nodata.counter_encoding::<u32>();
-        let mask = (1 << nbits) - 1;
+        let mask = (1u32 << nbits).saturating_sub(1);
```

Comment:
Is this to remove overflow checks?

Reply:
Yeah, I was hoping we could completely get rid of panics --- unfortunately, we can't, as the generated `idol` server code has some potential panic paths, but eliminating overflow checks here does shave a few bytes off the flash size regardless.